tests/test_rat_board.py

```diff
@@ -262,3 +262,39 @@ def test_next_free_mixed_board(self, keys: list, expected: int, rat_board_fx: Ra
 
         # and verify its functionality
         assert expected == nextFree
+
+    @pytest.mark.parametrize("test_input", [
+                            "UNIT_TEST",
+                            "@12345678-9876-53d1-ea5e-0000deadbeef",
+                            "42"])
+    def test_search_valid(self, test_input, rat_board_fx: RatBoard, rescue_plain_fx: Rescue):
+        rescue_plain_fx.uuid = UUID('12345678-9876-53d1-ea5e-0000deadbeef')
+        rat_board_fx.append(rescue_plain_fx)
+        assert rat_board_fx.search(test_input) == rescue_plain_fx
+
+    @pytest.mark.parametrize("test_input", [
+                            "unit_tes",  # bad client name
+                            "42.5",  # bad case number (decimals)
+                            "100",  # bad case number (int)
+                            '@12345678-9876-53d1-ea5e-0000dead'  # wrong length uuid
+                            ])
+    def test_search_garbage_returns_none(self, test_input, rat_board_fx: RatBoard,
+                                         rescue_plain_fx: Rescue):
+        rescue_plain_fx.uuid = UUID('12345678-9876-53d1-ea5e-0000deadbeef')
+        rat_board_fx.append(rescue_plain_fx)
+        assert rat_board_fx.search(test_input) is None
+
+    @pytest.mark.parametrize("test_input", [
+                            '@12345678-9876-53d1-ea5e-0000deadbee-',
+                            '@12345678-9876-53d1-ea5e-000deadsheep',
+                            True,
+                            False,
+                            42,
+                            uuid4(),
+                            None])
+    def test_search_garbage_raises_exception(self, test_input, rat_board_fx: RatBoard,
+                                             rescue_plain_fx: Rescue):
+        rescue_plain_fx.uuid = UUID('12345678-9876-53d1-ea5e-0000deadbeef')
+        rat_board_fx.append(rescue_plain_fx)
+        with pytest.raises((TypeError, ValueError,)):
+            assert rat_board_fx.search(test_input) is None
```

Comment:
Couldn't you just call .search directly, and not assert its output? as it doesn't output because its expected to raise?